docs/docs/development/hardware-integration/physical-layouts.md

```diff
@@ -0,0 +1,429 @@
+
+```dts
+numpad_map: numpad {
+    physical-layout = <&numpad_layout>;
+    positions
+        = < 0  1  2  3>
+        , < 4  5  6  7>
+        , < 8  9 10   >
+        , <11 12 13 14>
+        , <15    16   >;
+};
+```
+
+As a reminder, the `positions` property is a one-dimensional array like the `keys` property, formatted nicely through the use of whitespace and angle bracket groupings.
+
+If the left side with more keys was used as the reference layout, then the overall position map of the keyboard would look like this:
+
+```dts
+/ {
+    keypad_lossless_position_map {
+        compatible = "zmk,physical-layout-position-map";
+        complete;
+
+        macropad_map: macropad {
+            physical-layout = <&macropad_layout>;
+            positions
+                = < 0  1  2  3>
+                , < 4  5  6  7>
+                , < 8  9 10 11>
+                , <12 13 14 15>
+                , <16 17 18 19>;
+        };
+
+        numpad_map: numpad {
+            physical-layout = <&numpad_layout>;
+            positions
+                = < 0  1  2  3>
+                , < 4  5  6  7>
+                , < 8  9 10 17>
+                , <11 12 13 14>
+                , <15 18 16 19>;
+        };
+    };
+};
+```
+
+The "missing" positions are filled with the "spare" numbers of the layout with more keys. Meanwhile, if the right side with fewer keys is used as a reference, then the overall position map would look like this:
+
+```dts
+/ {
+    keypad_lossy_position_map {
+        compatible = "zmk,physical-layout-position-map";
+        complete;
+
+        macropad_map: macropad {
+            physical-layout = <&macropad_layout>;
+            positions
+                = < 0  1  2  3>
+                , < 4  5  6  7>
+                , < 8  9 10   >
+                , <12 13 14 15>
+                , <16    18   >;
+        };
+
+        numpad_map: numpad {
+            physical-layout = <&numpad_layout>;
+            positions
+                = < 0  1  2  3>
+                , < 4  5  6  7>
+                , < 8  9 10   >
+                , <11 12 13 14>
+                , <15    16   >;
+        };
+    };
+};
+```
+
+The above example is "lossy" because (unlike the "lossless" example) if a user switches from the macropad layout to the numpad layout _and then_ switches from the numpad layout back to the macropad layout, the assignments to the keys present but not listed in the macropad's map are lost.
+
+#### Example non-`complete` position map
+
+Consider the above device again - most of the positions have identical `keys` properties. For example, the numpad's `15` key and the macropad's `16` key would have the same physical property, and be mapped to each other automatically. The keys whose mappings are unable to be determined automatically are those with different physical characteristics - the 2u high and 2u wide keys, and their corresponding 1u counterparts.
```

Comment:
```suggestion
Consider the above device again -- most of the positions have identical `keys` properties. For example, the macropad's `0` key and the numpad's `0` key would have the same physical property, and be mapped to each other automatically. The keys whose mappings are unable to be determined automatically are those with different physical characteristics: the 2u high and 2u wide keys, and their corresponding 1u counterparts.
```
While I believe the previous version was technically true, the width of 16/15 were actually different so it had to be qualified. This is probably a simpler example. (also switched order to obey left/right)